src/main/java/com/salesforce/dataloader/action/visitor/BulkQueryVisitor.java

```diff
@@ -100,9 +104,33 @@ private void writeExtractionForBatch(BatchInfo batch) throws AsyncApiException,
 
         for (final String resultId : results.getResult()) {
             if (getProgressMonitor().isCanceled()) return;
+            boolean bufferResults = getController().getConfig().getBoolean(Config.BUFFER_UNPROCESSED_BULK_QUERY_RESULTS);
```

Comment:
Does this work?
```suggestion
            boolean bufferResults = getConfig().getBoolean(Config.BUFFER_UNPROCESSED_BULK_QUERY_RESULTS);
```

Reply:
Also, do you need to call it for every `for` iteration?

Reply:
another good catch (2 of them). I will make the change.